### ISSUE TITLE
Add Content on Different ML Model

### ISSUE THREAD

**Issue by Shouryabhardwajj** [animator/learn-python#856]
Description
Linear Regression
Logistic Regression
Decision Trees
Random Forests
Support Vector Machines (SVM)
K-Nearest Neighbors (KNN)
Naive Bayes
Neural Networks (including Deep Learning models)
Gradient Boosting Machines (GBM)
XGBoost
LightGBM
CatBoost
Principal Component Analysis (PCA)
K-Means Clustering
Hierarchical Clustering
Association Rule Learning (e.g., Apriori algorithm)
Gaussian Mixture Models (GMM)
Hidden Markov Models (HMM)
Time Series Models (e.g., ARIMA)
Reinforcement Learning Algorithms (e.g., Q-Learning)
Ensemble Methods (e.g., Stacking, Bagging)
Transfer Learning Models

@animator Please assign me this issue
Can you please label it as level 3 as I will be providing lot of content
